src/riscv/lib/src/state_backend/verify_backend.rs

```diff
@@ -96,6 +96,51 @@ impl ManagerBase for Verifier {
     }
 }
 
+#[cfg(test)]
+mod test_helpers {
+    use crate::state_backend::ManagerAlloc;
+    use crate::state_backend::verify_backend::DynRegion;
+    use crate::state_backend::verify_backend::PageId;
+    use crate::state_backend::verify_backend::Region;
+    use crate::state_backend::verify_backend::Verifier;
+
+    /// Trait for zero-initialized dynamic regions.
+    ///
+    /// Zero-initialising a dynamic region requires computing the  constant `NR_PAGES` which can not
+    /// be evaluated in a function. As a workaround we add it as an associated constant to this
+    /// trait.
+    trait ZeroInit<const LEAF_SIZE: usize, const LEN: usize> {
+        /// Number of pages in the dynamic region.
+        const NR_PAGES: usize = LEN.div_euclid(LEAF_SIZE);
```

Comment:
You need to round up.

```suggestion
        const NR_PAGES: usize = LEN.div_ceil(LEAF_SIZE);
```

Reply:
Also, where is this constant used through the trait?

Reply:
It's used in the `zero_initialized` function. I can not use its expression inside the function body sadly, so I have to use an associated constant.

Reply:
I tried creating a `const` and that's why it wasn't working 🤦 . With just a `let` I can get rid of the trait. [diff](https://github.com/tezos/riscv-pvm/compare/5cbff6e72fde3382f86ff8290f9f7e5eaf37cb34..81849a835d4dd4987e573148bd9a6eed34d0d195)